GetStream/stream-chat-swift
# [CIS-1455] Pinned messages pagination support

### 🔗 Issue Link

CIS-1455

### 🎯 Goal

1. Make it possible to paginate messages pinned in a channel
2. Make it possible to change the pinning date

It is worth mentioning that saving pinned messages to the database causes gaps and breaks pagination in the main channel. The PR contains changes allowing to create `ChatMessage` model directly from `MessagePayload` without saving it to the database first.

### 🛠 Implementation

- Add endpoint to fetch messages pinned in a channel
- Expose `loadPinnedMessages` API on channel controller
- Make it possible to change pinning date

### 🧪 Testing

Changes are covered by unit-tests.

Manual testing:
1. Checkout `origin/pinned-messages-pagination` branch
2. Launch `DemoApp`
3. Sign in as `C-3PO`
4. Open the channel, pin/unpin messages
5. Navigate to `PinnedMessagesVC` and see pagination is correct

### ☑️ Checklist
